File: teklif-api/src/main/java/com/trafik/teklif_api/service/SystemSettingsService.java
package com.trafik.teklif_api.service;


import com.trafik.teklif_api.dto.settings.SystemSettingsRequest;
import com.trafik.teklif_api.dto.settings.SystemSettingsResponse;

public interface SystemSettingsService {
    SystemSettingsResponse get();
    void update(SystemSettingsRequest req);
}

File: frontend/project/src/services/SystemSettingsService.ts
// src/services/SystemSettingsService.ts
import api from '../api/axios';

export type SystemSettingsResponse = {
  agencyName: string | null;
  agencyCode: string | null;
  licenseNumber: string | null;
  joinDate: string | null;   // yyyy-MM-dd
  lastLogin: string | null;  // yyyy-MM-dd veya ISO
  isActive: boolean | null;
};

export type SystemSettingsRequest = {
  agencyName?: string | null;
  agencyCode?: string | null;
  licenseNumber?: string | null;
  joinDate?: string | null;
  lastLogin?: string | null;
  isActive?: boolean | null;
};

export const SystemSettingsService = {
  get: async () => (await api.get<SystemSettingsResponse>('/system/settings')).data,
  update: async (body: SystemSettingsRequest) => {
    await api.put('/system/settings', body);
  },
};
